core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -197,14 +207,7 @@ class ZkUtils(val zkClient: ZkClient,
                               BrokerSequenceIdPath,
                               IsrChangeNotificationPath)
 
-  val securePersistentZkPaths = Seq(BrokerIdsPath,
```

Comment:
I'm confused, you're commenting on old code. 
`getEntityConfigRootPath(ConfigType.Topic)`
and `getEntityConfigRootPath(ConfigType.Broker)`
all produce the `/config/<>` path so that's why you can see I've include `/config` here:
https://github.com/apache/kafka/pull/2655/files#diff-f4fe18d654e20048317014cfe9d53968R62

Secondly, if I give the parent, the function I believe does all the recursion for me, see here:  https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala#L146